src/util.cpp

```diff
@@ -423,16 +432,14 @@ void ReleaseDirectoryLocks()
 /** Interpret string as boolean, for argument parsing */
 static bool InterpretBool(const std::string& strValue)
 {
-    if (strValue.empty())
-        return true;
-    return (atoi(strValue) != 0);
+    // These values mean false, anything else (including empty string) is true.
+    return strValue != "0" && strValue != "false";
```

Comment:
I think this has nothing to do with `printtoconsole`. Mind to split it up into a separate pull?

Reply:
It has to do with the behavior I wanted for disabling the debug log, but you're right that it's technically orthogonal. I will split it out.